src/org/zaproxy/zap/extension/api/CoreAPI.java

```diff
@@ -926,13 +930,29 @@ public ApiResponse handleApiView(String name, JSONObject params)
 				((ApiResponseList)result).addItem(new ApiResponseElement("host", site));
 			}
 		} else if (VIEW_SITES.equals(name)) {
-			result = new ApiResponseList(name);
-			SiteNode root = (SiteNode) session.getSiteTree().getRoot();
-			@SuppressWarnings("unchecked")
-			Enumeration<TreeNode> en = root.children();
-			while (en.hasMoreElements()) {
-				((ApiResponseList)result).addItem(new ApiResponseElement("site", ((SiteNode) en.nextElement()).getNodeName()));
-			}
+            result = new ApiResponseList(name);
+            if (!Constant.isLowMemoryOptionSet()) {
+                SiteNode root = (SiteNode) session.getSiteTree().getRoot();
+                @SuppressWarnings("unchecked")
+                Enumeration<TreeNode> en = root.children();
+                while (en.hasMoreElements()) {
+                    ((ApiResponseList)result).addItem(new ApiResponseElement("site", ((SiteNode) en.nextElement()).getNodeName()));
+                }
+            }else{
+                try {
+                    Model model = Model.getSingleton();
+                    RecordStructure rs = model.getDb().getTableStructure().find(session.getSessionId(), "Root", "GET");
+                    StructuralNode rootStructuralNode = new StructuralTableNode(rs);
+                    // ((ApiResponseList)result).addItem(new ApiResponseElement("site", rootStructuralNode.getName()));
+                    
+                    for (Iterator<StructuralNode> iterator = rootStructuralNode.getChildIterator(); iterator.hasNext();) {
+                        StructuralNode child = iterator.next();
+                        ((ApiResponseList)result).addItem(new ApiResponseElement("site", child.getName()));
+                    }
+                } catch (DatabaseException e) {
+                    logger.debug(e.getMessage(), e);
```

Comment:
This should be logged as error (and throw an `ApiException` with type `INTERNAL_ERROR`).